### DISCUSSION TITLE
why pastel-powerline theme work fine on vs code terminal but not in mac terminal

### DISCUSSION THREAD

**Discussion by serene-ding** [starship/starship#5413] [Q&A]
i have installed fira nerd font and then installed pastel-powerline theme but the theme works fine for vscode terminal, not for mac terminal. I have already change mac terminal font to fira nerd font
![image](https://github.com/starship/starship/assets/77049532/e1a27578-dcea-46ee-a4eb-b993cb43de9e)
![image](https://github.com/starship/starship/assets/77049532/1aebb3cf-71ba-4b09-88c9-ff7f187a871c)

**Comment by davidkna** [comment]
`Terminal.app` does not support RGB-colors (`#aabbcc`). You need to use named colors or the numbered colors.

**Reply by serene-ding** [reply]
so i need to edit the toml json file of the theme and change rgb color to named color?

**Reply by davidkna** [reply]
It should work with named colors.

**Reply by mickimnet** [reply]
Alternatively you could use [iTerm2](https://iterm2.com), which does support _true color_, as well as a lot of other improvements over _Terminal.app_.

**Reply by serene-ding** [reply]
in iterm2 the color works right but the layout is a mess.
![image](https://github.com/starship/starship/assets/77049532/20be5093-3266-4db3-bf96-ed9285f3787a)

**Reply by mickimnet** [reply]
What font did you use in _iTerm_? Should be a _Nerd Font_ as well.

**Comment by mcandre** [comment]
Would be helpful for the presets to be more portable out of the box.

Also, please don't overwrite the entire configuration when applying a preset. Append.

**Comment by davidkna** [comment]
As an update: Since macOS 26 the colours of the preset should work in Terminal.app

**Comment by stella-sre** [comment]
The pastel-powerline theme uses **Powerline glyphs** and **Nerd Font icons** that require special fonts.

**VS Code works** because it has built-in Nerd Font support.

**Mac Terminal.app fails** because it uses system fonts lacking these glyphs.

## How To Fix It

### Step 1: Install a Nerd Font

```bash
brew tap homebrew/cask-fonts
brew install --cask font-meslo-lg-nerd-font
```

Or download from: https://www.nerdfonts.com/font-downloads

Recommended fonts:

- MesloLGS NF
- JetBrainsMono Nerd Font
- FiraCode Nerd Font

### Step 2: Configure Terminal.app

1. Open **Terminal** → **Preferences** (⌘ + ,)
2. Go to **Profiles** → **Text** tab
3. Click **Change** font
4. Select your Nerd Font (e.g., "MesloLGS NF")

### Step 3: Consider switching terminals

Mac Terminal.app has limitations. Better options:

- **iTerm2** — https://iterm2.com/
- **Kitty** — https://sw.kovidgoyal.net/kitty/
- **Alacritty** — https://alacritty.org/

## Additional Recommendations

1. **Verify font is installed:**

   ```bash
   fc-list | grep -i "meslo"
   ```

2. **Test Nerd Font glyphs:**

   ```bash
   echo -e "\ue0b0 \ue0b2 \uf113 \uf292"
   ```

   You should see: arrow shapes and icons

3. **If still broken**, try a simpler preset:
   ```bash
   starship preset plain-text -o ~/.config/starship.toml
   ```

---

Let me know if you need help configuring iTerm2!
